src/controllers/courseController.ts

```diff
@@ -162,6 +165,23 @@ export class CourseController {
 		}
 	}
 
+	setStatus() {
+		return async (request: Request, response: Response) => {
+			const errors = await this.courseValidator.check(request.body, ['status'])
+
+			if (errors.size) {
+				request.session!.sessionFlash = {errors: errors}
```

Comment:
If the session is not explicitly saved before redirection there's a race condition between node.js to save the session and client browser making redirect request, this is explained here: https://baudehlo.com/2016/02/19/express-lost-sessions-and-windows/, the blog post talks about Windows but the OS in to really relevant so much, rather the race condition pattern